### DISCUSSION TITLE
No C compiler found! "cc", "gcc", "clang", "cl", "zig" are not executable.

### DISCUSSION THREAD

**Discussion by gerroon** [LazyVim/LazyVim#1920] [Q&A] [answered]
Hi

I  installed it on Win 11. I do not want to install compilers since i do not need anything compiler stuff. Is there a way top disable this error or the functionality causing this error?

`Error detected while processing User Autocommands for "VeryLazy":
No C compiler found! "cc", "gcc", "clang", "cl", "zig" are not executable.`

thanks

**Comment by amaanq** [comment]
Tree-sitter needs a compiler, feel free to disable it.

**Comment by gerroon** [comment]
How about    mason-lspconfig lua server issue? I disabled mason I still get an error when opening a file :(


`[mason-lspconfig.nvim] Server "lua_ls" is not a valid entry in ensure_installed. Make sure to only provide lspconfig server names.`


```
return {
  -- disable trouble
  { "nvim-treesitter", enabled = false },
  { "nvim-ts-autotag", enabled = false },
  { "nvim-treesitter-context", enabled = false },
  { "mason-lspconfig", enabled = false },
  { "mason", enabled = false },
}

```

**Reply by gerroon** [reply]
How do I disable Mason lsp config?

**Reply by dpetka2001** [reply]
What is your configuration for `mason-lspconfig`?

**Reply by gerroon** [reply]
I do not know, I never set it up. It comes with LazyVim I think, I never need to use it myself. 

I put these in disabled.lua but it does not disable it. I am a new user btw.

```
return {
  -- disable trouble
  { "nvim-treesitter", enabled = false },
  { "nvim-ts-autotag", enabled = false },
  { "nvim-treesitter-context", enabled = false },
  { "mason-lspconfig", enabled = false },
  { "mason", enabled = false },
}


```

**Reply by dpetka2001** [reply, marked as answer]
It should be 
```lua
return {
  { "williamboman/mason-lspconfig.nvim", enabled = false },
}
```
With the above code it gets disabled for me and then in Lazy UI you just have to press `X` to clean it up from Lazy.

**Comment by hros** [comment]
> It should be
> 
> ```lua
> return {
>   { "williamboman/mason-lspconfig.nvim", enabled = false },
> }
> ```
> 
> With the above code it gets disabled for me and then in Lazy UI you just have to press `X` to clean it up from Lazy.

1. in what file do you add the above line?
2. Are any features lost when I disable this?

**Reply by dusty-phillips** [reply]
It can go in any lua file in the plugins directory.  I have a file named disabled.lua You can read through the mason-lspconfig readme to understand what features the plugin brings.

**Reply by hros** [reply]
thanks @dusty-phillips ,
while waiting for your answer, I installed `zig` as it is a small package
I used `winget install -e --id zig.zig`, but this results in `neovim` not finding the `zig` environment (as described in [this ](https://github.com/ziglang/zig/issues/16670)issue)
how do I get `nvim` to ignore the installed (and unfunctional) `zig` compiler?

**Comment by philharmonie** [comment]
I am running into the same issue with a fresh nvim installation on a nixOS system. I cloned lazyvim and on the initial opening, I get the error:
```
Error detected while processing User Autocommands for "VeryLazy":
No C compiler found! "cc", "gcc", "clang", "cl", "zig" are not executable.
Press ENTER or type command to continue
```
_By the way: pressing ENTER just prints the error again_.

As described, I created the `~/.config/nvim/lua/plugins/disabled.lua` with this content:
```lua
return {
  { "williamboman/mason-lspconfig.nvim", enabled = false },
}
```

This does not change anything noticeable.

**Reply by hros** [reply]
The simplest solution was to install the Zig compiler since it is very lightweight
There is a [discussion](https://ziggit.dev/t/build-and-use-the-latest-zig-using-nix/543/3) on using Zig on Nix
